### PR TITLE
add docker security warning

### PR DESCRIPTION
**Pull request by Kriechi**
- [ ] manually update the README at https://hub.docker.com/r/mitmproxy/mitmproxy/ after merging

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -5,8 +5,8 @@ mitmproxy
 
 This repository contains the **mitmproxy** and **pathod** projects.
 
-``mitmproxy`` is an interactive, SSL-capable intercepting proxy with a console
-interface.
+``mitmproxy`` is an interactive, TLS/SSL-capable intercepting proxy with a console
+interface for HTTP/1, HTTP/2, and WebSockets.
```
Comment (mhils) [inline README.rst]: -1 to the "for HTTP/1, HTTP/2, and WebSockets." addition, IMHO the value/additional characters ratio isn't high enough.
Also I think we should either do "TLS" or "SSL/TLS".

File: docs/src/content/_index.md [hunk 2]
```diff
@@ -11,8 +11,7 @@ menu:
 The mitmproxy project's tools are a set of front-ends that expose common
 underlying functionality.
 
-**mitmproxy** is an interactive man-in-the-middle proxy for HTTP and HTTPS
-with a console interface.
+**mitmproxy** is an interactive, TLS/SSL-capable intercepting proxy with a console interface for HTTP/1, HTTP/2, and WebSockets.
```
Comment (mhils) [inline docs/src/content/_index.md]: (see above)

File: release/docker/README.md [hunk 3]
```diff
@@ -1,6 +1,6 @@
 # mitmproxy
 
-Containerized version of [mitmproxy](https://mitmproxy.org/), an interactive SSL-capable intercepting HTTP proxy.
+Containerized version of [mitmproxy](https://mitmproxy.org/), an interactive, TLS/SSL-capable intercepting proxy for HTTP/1, HTTP/2, and WebSockets.
```
Comment (mhils) [inline release/docker/README.md]: (see above)

**Comment by Kriechi** [comment]
Right - I split the readme changes to #3329 to allow for more discussion.
This PR now only contains the Docker security warning changes and moving of docker-related files to `release/`.

File: release/docker/README.md [hunk 4]
```diff
@@ -36,3 +36,12 @@ For further details, please consult the mitmproxy [documentation](http://docs.mi
 # Tags
 
 The available release tags can be seen [here](https://hub.docker.com/r/mitmproxy/mitmproxy/tags/).
+
+* `master` always tracks the git-master branch and represents the unstable development tree.
+* `latest` always points to the same image as the most recent stable release, including bugfix releases (e.g., `4.0.0` and `4.0.1`).
+* `X.Y.Z` tags contain the mitmproxy release with this version number.  
+  - All packages, dependencies, and the base image that were available at the time of the mitmproxy release.
```
Comment (mhils) [inline release/docker/README.md]: We have the following at https://docs.mitmproxy.org/stable/overview-installation/#self-contained-pre-built-binary-packages:

> Dependencies in the binary packages are frozen on release, and can’t be updated in situ. This means that we necessarily capture any bugs or security issues that may be present. We don’t generally release new binary packages simply to update dependencies (though we may do so if we become aware of a really serious issue). If you use our binary packages, please make sure you update regularly to ensure that everything remains current.

Do we maybe want to re-use the same wording for Docker (s/binary packages/docker images/)?

**Comment by Kriechi** [comment]
FYI: We have to manually copy & paste changes to the README over at the Docker Hub page - it is NOT auto-updating.
https://hub.docker.com/r/mitmproxy/mitmproxy/

**Pull request merged**